### PR TITLE
Support for parameters with long data types

### PR DESCRIPTION
**Pull request by edharper01**
Unlike freeTDS, the MS ODBC driver (on both Windows and Linux) provides proper support for long data types - varchar/nvarchar/varbinary(max) - where the value is bigger than 8kb.

This change amends the ExtractParameters method to detect these parameters based on the size property.

### CODE REVIEW

File: column.go [hunk 1]
```diff
@@ -11,7 +11,7 @@ import (
 	"time"
 	"unsafe"
 
-	"github.com/alexbrainman/odbc/api"
+	"github.com/edharper01/odbc/api"
```
Comment (alexbrainman) [inline column.go]: You shouldn't rename import path. Otherwise users of this package will end-up using source from your github repo.

See, for example, these
https://splice.com/blog/contributing-open-source-git-repositories-go/
http://blog.campoy.cat/2014/03/github-and-go-forking-pull-requests-and.html
on how to create pull request without changing import path.

File: api/api_unix.go [hunk 2]
```diff
@@ -48,6 +48,8 @@ const (
 	SQL_NULL_HSTMT         = uintptr(C.SQL_NULL_HSTMT)
 
 	SQL_PARAM_INPUT = C.SQL_PARAM_INPUT
+	SQL_PARAM_INPUT_OUTPUT = C.SQL_PARAM_INPUT_OUTPUT
+	SQL_PARAM_OUTPUT = C.SQL_PARAM_OUTPUT
```
Comment (alexbrainman) [inline api/api_unix.go]: I don't see why change this file.

File: api/api_windows.go [hunk 3]
```diff
@@ -33,6 +33,8 @@ const (
 	SQL_NULL_HSTMT         = 0
 
 	SQL_PARAM_INPUT = 1
+	SQL_PARAM_INPUT_OUTPUT = 2
+	SQL_PARAM_OUTPUT = 4
```
Comment (alexbrainman) [inline api/api_windows.go]: I don't see why change this file.

File: error.go [hunk 4]
```diff
@@ -24,7 +24,7 @@ type DiagRecord struct {
 }
 
 func (r *DiagRecord) String() string {
-	return fmt.Sprintf("{%s} %s", r.State, r.Message)
+	return fmt.Sprintf("{%d} - {%s} %s", r.NativeError,r.State, r.Message)
```
Comment (alexbrainman) [inline error.go]: This is unrelated to the fix. Please revert this.

File: param.go [hunk 5]
```diff
@@ -174,6 +174,20 @@ func ExtractParameters(h api.SQLHSTMT) ([]Parameter, error) {
...
+			switch p.SQLType {
+				case api.SQL_VARBINARY:
+					p.SQLType = api.SQL_LONGVARBINARY
+				case api.SQL_VARCHAR:
+					p.SQLType = api.SQL_LONGVARCHAR
+				case api.SQL_WVARCHAR:
+					p.SQLType = api.SQL_WLONGVARCHAR
+			}
+		}
+
```
Comment (alexbrainman) [inline param.go]: Remove empty line.

File: param.go [hunk 6]
```diff
@@ -174,6 +174,20 @@ func ExtractParameters(h api.SQLHSTMT) ([]Parameter, error) {
 			continue
 		}
 		p.isDescribed = true
+
```
Comment (alexbrainman) [inline param.go]: Remove empty line.

File: param.go [hunk 7]
```diff
@@ -174,6 +174,20 @@ func ExtractParameters(h api.SQLHSTMT) ([]Parameter, error) {
 			continue
 		}
 		p.isDescribed = true
+		
+		// EJH - SQL Server MAX types (varchar(max), nvarchar(max),
```
Comment (alexbrainman) [inline param.go]: What does EJH stands for? Remove.

**Comment by alexbrainman** [comment]
Also, please, squash all commits into one.
Otherwise looks good. Thank you very much.

Alex

**Comment by edharper01** [comment]
Hopefully I've made all the changes you asked for; thanks for the very
helpful links about how to work with forks.

**Comment by alexbrainman** [comment]
Please go fmt the code, and I will submit. Thank you.

Alex

**Comment by edharper01** [comment]
The change should be good to go now - I have applied gofmt to param.go
Thanks for you patience.
Ed

**Comment by alexbrainman** [comment]
> I have applied gofmt to param.go

You can just run `go fmt` and it will format ALL you code in the current directory. You can also do `go fmt .\...` to format everything in the current directory including all sub-directories.

Alex
